homeassistant/components/environment_canada/camera.py

```diff
@@ -35,19 +52,17 @@ def __init__(self, coordinator):
         self._attr_entity_registry_enabled_default = False
 
         self.content_type = "image/gif"
-        self.image = None
-        self.observation_time = None
 
     def camera_image(
         self, width: int | None = None, height: int | None = None
     ) -> bytes | None:
         """Return bytes of camera image."""
-        if not hasattr(self.radar_object, "timestamp"):
-            return None
-        self.observation_time = self.radar_object.timestamp
+        self._attr_extra_state_attributes = {
+            ATTR_OBSERVATION_TIME: self.radar_object.timestamp,
+        }
```

Comment:
This is existing, but it really should be its own timestamp sensor instead.  Its going to create a new row in the attributes table each time it changes along with all the other camera attributes since it can't be de-duplicated.

Reply:
will look at converting to sensor. already have sensor for the forecast last update, so this would be similar.